### PR TITLE
Ulm

### PR DESCRIPTION
**Pull request by Zemurin**

### CODE REVIEW

**Comment by Zemurin** [comment]
Most of the changes are just busywork. Only added logic was to handle dynamic-* cultures which don't have mappings and their diverged neocultures.

File: EU4toV2/Source/V2World/Output/outCultureGroups.cpp [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #include "CultureGroups/CultureGroup.h"
-#include "CultureGroups/CultureGroups.h"
+#include "CultureGroups/CultureGroupsManager.h"
 #include "output.h"
 
-std::ostream& mappers::operator<<(std::ostream& output, const CultureGroups& cultureGroupsMapper)
+std::ostream& mappers::operator<<(std::ostream& output, const CultureGroupsManager& cultureGroupsMapper)
```
Comment (Idhrendur) [inline EU4toV2/Source/V2World/Output/outCultureGroups.cpp:5]:
```suggestion
std::ostream& mappers::operator<<(std::ostream& output, const CultureGroupsManager& cultureGroupsManager)
```

File: EU4toV2/Source/Mappers/CultureGroupsMapper/CultureGroupsMapping.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+void mappers::CultureGroupsMapping::registerKeys()
+{
+	registerKeyword("v2", [this](std::istream& theStream) {
+		v2 = commonItems::getString(theStream);
+	});
+	registerKeyword("hpm", [this](std::istream& theStream) {
+		hpm = commonItems::getString(theStream);
+		if (theConfiguration.isHpmEnabled())
```
Comment (Idhrendur) [inline EU4toV2/Source/Mappers/CultureGroupsMapper/CultureGroupsMapping.cpp:20]: Is the configuration coming in as a global? It'd be clearer if a boolean was passed in here instead.

File: EU4toV2/Source/Mappers/CultureGroupsMapper/CultureGroupsMapper.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	registerKeys();
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+void mappers::CultureGroupsMapper::registerKeys()
+{
+	registerKeyword("link", [this](std::istream& theStream) {
+		const CultureGroupsMapping mapping(theStream);
+		if (!mapping.getV2().empty() && !mapping.getEU4s().empty())
```
Comment (Idhrendur) [inline EU4toV2/Source/Mappers/CultureGroupsMapper/CultureGroupsMapper.cpp:27]: Checking for eu4 mappings is redundant here. The loop below will do nothing if its empty.

**Review by Idhrendur — APPROVED** [review]

**Pull request merged**